lib/kadm5/chpass_c.c

```diff
@@ -62,33 +62,45 @@ kadm5_c_chpass_principal(void *server_handle,
     if(ret)
 	return ret;
 
+    krb5_data_zero(&reply);
+
     sp = krb5_storage_from_mem(buf, sizeof(buf));
     if (sp == NULL) {
-	krb5_clear_error_message(context->context);
-	return ENOMEM;
+	ret = ENOMEM;
+	goto out;
     }
-    krb5_store_int32(sp, kadm_chpass);
-    krb5_store_principal(sp, princ);
-    krb5_store_string(sp, password);
-    krb5_store_int32(sp, keepold); /* extension */
+    ret = krb5_store_int32(sp, kadm_chpass);
+    if (ret)
+	goto out;
+    ret = krb5_store_principal(sp, princ);
+    if (ret)
+	goto out;
+    ret = krb5_store_string(sp, password);
+    if (ret)
+	goto out;
+    ret = krb5_store_int32(sp, keepold); /* extension */
+    if (ret)
+	goto out;
```

Comment:
How?

Reply:
On failure the cascade style requires that `if (ret == 0)` be compared for each statement that will be skipped.

Reply:
That's not an optimization we need though.

Reply:
@nicowilliams As I started to convert to your preferred style it ends up being problematic because some of the errors we want to return after clearing the krb5_context error state and others we do not.   

I'm updating the patch for the krb5_enomem() changes but otherwise leaving it as-is.

Reply:
Ok.